### PR TITLE
Updates for use of observing sessions

### PR DESCRIPTION
**Pull request by tskisner**
This has changes needed to support the use of the existing observing sessions across the code base.  It also fixes several bugs discovered in that process.

* SimGround operator now has a trait specifying the input focalplane key to use for splitting telescopes into observations within one session.  Move some code out of exec() and into a separate method.

* In atmosphere simulations, group detectors by frequency range and separately calculate the common absorption and loading for each.

* Use session UID instead of observation UID across operators where that was the intended behavior.

* Support multifrequency fake focalplanes for unit testing.

* When simulating noise models whose streams are detector PSDs, use the detector UID for the index.

* Fix typo in Bandpass.get_range() method.

* Fix bug introduced recently in Poly2D filter.

* Fix unit test for YieldCut operator to account for case of running with MPI where each process has a subset of the global detectors.

### CODE REVIEW

**Comment by tskisner** [comment]
Note that I am still working on resolving an MPI unit test failure on the Poly2D filter.

File: src/libtoast/src/toast_tod_filter.cpp [hunk 1]
```diff
@@ -378,7 +378,7 @@ void toast::filter_poly2D_solve(
         int rank;
         int info;
         int one = 1;
-        double rcond_limit = 1e-3;
+        double rcond_limit = 1e-6;
```
Comment (tskisner) [inline src/libtoast/src/toast_tod_filter.cpp:381]: Eventually we could make this a parameter that is passed down from the python layer.

File: src/toast/instrument.py [hunk 2]
```diff
@@ -326,7 +326,7 @@ def __init__(self, bandcenters, bandwidths, nstep=101):
     def get_range(self, det=None):
         """Return the maximum range of frequencies needed for convolution."""
         if det is not None:
-            return self.fmin[det], self.fmin[det]
+            return self.fmin[det], self.fmax[det]
```
Comment (tskisner) [inline src/toast/instrument.py:329]: This is fixing an unrelated typo.

File: src/toast/ops/polyfilter.py [hunk 3]
```diff
@@ -333,7 +332,7 @@ def _exec(self, data, detectors=None, **kwargs):
                             proj = np.dot(t, signals[isample, good] * mask)
                             ccinv = np.dot(t, t.T)
                             coeff[isample, igroup] = np.linalg.lstsq(
-                                ccinv, proj, rcond=None
+                                ccinv, proj, rcond=1.0e-6
```
Comment (tskisner) [inline src/toast/ops/polyfilter.py:335]: For consistency with the compiled code for now.

File: src/toast/ops/sim_tod_atm_generate.py [hunk 4]
```diff
@@ -247,18 +245,18 @@ def _exec(self, data, detectors=None, **kwargs):
             elmin = None
             elmax = None
             for ob in sdata.obs:
-                ob_azmin, ob_azmax, ob_elmin, ob_elmax = self._get_scan_range(first_obs)
+                ob_azmin, ob_azmax, ob_elmin, ob_elmax = self._get_scan_range(ob)
```
Comment (keskitalo) [inline src/toast/ops/sim_tod_atm_generate.py:248]: Nice catch!

**Review by keskitalo — APPROVED** [review]
Looks great. Only minor suggestions. I'll need to rebase my processing mask against `toast3` once this is merged. There will be conflicts.

File: src/toast/ops/sim_crosstalk.py [hunk 5]
```diff
@@ -190,7 +190,7 @@ def exec_roundrobin(self, data, detectors=None, **kwargs):
             exists = ob.detdata.ensure(
                 self.det_data, detectors=dets, create_units=self.det_data_units
             )
-            obsindx = ob.uid
+            sindx = ob.session.uid
```
[the remark this answers is not in the document]
Comment (tskisner) [reply src/toast/ops/sim_crosstalk.py]: Removed.

**Comment by keskitalo** [comment]
This PR should be rebased to resolve any potential conflicts from the processing mask merge.

**Pull request merged**